### PR TITLE
Companion for substrate#9732

### PR DESCRIPTION
**Pull request by cheme**
companion for https://github.com/paritytech/substrate/pull/9732

### CODE REVIEW

File: client/service/src/genesis.rs [hunk 1]
```diff
@@ -21,21 +21,26 @@ use sp_runtime::traits::{Block as BlockT, Hash as HashT, Header as HeaderT, Zero
 /// Generate the genesis block from a given ChainSpec.
 pub fn generate_genesis_block<Block: BlockT>(
 	chain_spec: &Box<dyn ChainSpec>,
+	genesis_state_version: sp_runtime::StateVersion,
```
Comment (bkchr) [inline client/service/src/genesis.rs:24]: Can we not fetch the runtime version from the code in the genesis state? As we do this in Substrate?

File: polkadot-parachains/seedling/src/lib.rs [hunk 2]
```diff
@@ -60,6 +60,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	impl_version: 0,
 	apis: RUNTIME_API_VERSIONS,
 	transaction_version: 1,
+	state_version: 1,
```
Comment (bkchr) [inline polkadot-parachains/seedling/src/lib.rs:63]:
```suggestion
	state_version: 0,
```

Probably too late.

File: client/service/src/genesis.rs (hunk 1 above)
Comment (cheme) [reply client/service/src/genesis.rs:24]: That was my first idea, but current code need an executor (for compatibility I think), so ends up being really easier to use this way.
You think it is a must have, or can we keep it this way (seems only use from commands at this point)?

Comment (bkchr) [reply client/service/src/genesis.rs:24]: Hmm, this is not really future proof.

**Review by bkchr — APPROVED** [review]

**Comment by cheme** [comment]
bot merge

**Pull request merged**